lib/kitchen_sink/list/index_by.ex

```diff
@@ -0,0 +1,20 @@
+defmodule KitchenSink.List.IndexBy do
+  @doc """
+  creates and indexed list from a list of maps using a key_function or
+  a path. if a path is given then the access module will be used.
+
+  This function is very similar to group_by, however the values will
+  not be grouped. there should be a 1-to-1 mapping for the key_fun to
+  item in the list.  Do not use this if it's possible that you could
```

Comment:
um, the algorithm will still work, but dupe keys will be dropped.
i think raise is a little too much. my use-case may not be someone elses.
The warning is because if you really don't know about your input, and don't want to lose data, then group_by is the tool.

i think the raise you ask for would require a lot more steps.